src/V3Randomize.cpp

```diff
@@ -799,11 +809,12 @@ class ConstraintExprVisitor final : public VNVisitor {
                 std::string fmt;
                 // Normalize to standard bit width
                 if (actual_width <= 8) {
-                    fmt = "#x%2x";
+                    m_structSel ? fmt = "%2x" : fmt = "#x%2x";
                 } else if (actual_width <= 16) {
-                    fmt = "#x%4x";
+                    m_structSel ? fmt = "%4x" : fmt = "#x%4x";
```

Comment:
```suggestion
                    fmt = m_structSel ? "%4x" : "#x%4x";
```

Although is it actually important to have the correct width?  I think it's simpler if we just use %0x to get minimum widths.